extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -332,165 +336,171 @@ describe('ProjectsController: import operations', function (): void {
+		const spy = sinon.spy(vscode.window, 'showErrorMessage');
 
 		let testFolderPath = await testUtils.generateTestFolderPath();
 		testFolderPath += '_nonexistentFolder';	// Modify folder path to point to a nonexistent location
 
-		const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
+		const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
 
-		await testUtils.shouldThrowSpecificError(async () => await projController.generateList(testFolderPath), constants.cannotResolvePath(testFolderPath));
+		await projController.generateList(testFolderPath);
+		should(spy.calledOnce).be.true('showErrorMessage should have been called');
 	});
 
 	it('Should show error when no project name provided', async function (): Promise<void> {
 		for (const name of ['', '    ', undefined]) {
-			testContext.apiWrapper.reset();
-			testContext.apiWrapper.setup(x => x.showInputBox(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(name));
-			testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
+			sinon.stub(vscode.window, 'showInputBox').returns(Promise.resolve(name));
+			const spy = sinon.spy(vscode.window, 'showErrorMessage');
 
-			const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
-			await testUtils.shouldThrowSpecificError(async () => await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile }), constants.projectNameRequired, `case: '${name}'`);
+			const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
+			await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile });
+			should(spy.calledOnce).be.true('showErrorMessage should have been called');
+			sinon.restore();
 		}
 	});
 
 	it('Should show error when no target information provided', async function (): Promise<void> {
-		testContext.apiWrapper.setup(x => x.showInputBox(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve('MyProjectName'));
-		testContext.apiWrapper.setup(x => x.showQuickPick(TypeMoq.It.isAny())).returns(() => Promise.resolve(undefined));
-		testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
+		sinon.stub(vscode.window, 'showInputBox').returns(Promise.resolve('MyProjectName'));
+		sinon.stub(vscode.window, 'showQuickPick').returns(Promise.resolve(undefined));
+		const spy = sinon.spy(vscode.window, 'showErrorMessage');
 
-		const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
-		await testUtils.shouldThrowSpecificError(async () => await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile }), constants.extractTargetRequired);
```

Comment:
Is there a way to verify the exact error that was thrown by the test execution, instead of just counting the number of "spy" calls for "showErrorMessage"?

Reply:
Yes, with `calledWith(args)`. I opted not to do that here because I didn't see a ton of value in checking the exact error message displayed, but I'll take a look at getting those back in just so I'm not changing your tests' behavior more than is needed.

Reply:
Ok, updated with checks for the correct error messages